src/format_changeset.rs

```diff
@@ -36,7 +36,7 @@ pub fn format_changeset(f: &mut fmt::Formatter, changeset: &Changeset) -> fmt::R
                 }
             }
             Difference::Add(ref added) => {
-                match diffs.get(i - 1) {
+                match diffs.get(i.wrapping_sub(1)) {
```

Comment:
I'd prefer not to rely on the last index to be `None`, but instead handle `i=0` explicitly, e.g.
```rust
let prev = i.checked_sub(1).and_then(|x| diffs.get(x));
match prev {
```